### PR TITLE
[NEW] Audio Notification updated in sidebar

### LINKED ISSUE #7696: Add setting to allow audio notifications for All Messages
Just like we have for Desktop, Email, & Push - allow the user to select the trigger for audio notifications. Currently, it only triggers for Mentions. Allow the user to choose "Mentions" or "All Messages". Not sure we need the third option of "Nothing" as that's already available under "Audio"->"None".

This setting should be shown in the "Notifications" flex tab in every Room/DM as the other options are shown.

### PR DESCRIPTION
**Pull request by aditya19496**
@RocketChat/core 

Closes #7696

### CODE REVIEW

**Comment by aditya19496** [comment]
@ggazzo Is it fine now?

**Comment by ggazzo** [comment]
@aditya19496 perfect! thanks!

File: server/methods/saveUserPreferences.js [hunk 1]
```diff
@@ -50,11 +50,7 @@ Meteor.methods({
 			} else {
 				delete preferences.mergeChannels;
 			}
-			if (settings.unreadRoomsMode) {
```
Comment (ggazzo) [inline server/methods/saveUserPreferences.js]: @aditya19496  why did you delete this code?

**Comment by ggazzo** [comment]
@aditya19496 I`m afraid that is not working =/ I got this error `Exception in defer callback: TypeError: RocketChat.models.Subscriptions.updateUserSubscription is not a function` could you check for me plz?

**Comment by aditya19496** [comment]
yeah sure. It must be due to the changes made again. I will check it and let you know asap.

File: server/methods/saveUserPreferences.js (hunk 1 above)
Comment (aditya19496) [inline server/methods/saveUserPreferences.js]: well it must be due to changes made again.
I will check all the changes again and let you know when it will be working perfectly.

**Comment by aditya19496** [comment]
@ggazzo  Sorry, I was busy attending hackathon.
I will continue working from now on.

**Comment by aditya19496** [comment]
Well I don't know if anyone modified files before/after my pull request, its not working anymore.
It will take time to make it working again. I will let you know asap.

**Comment by aditya19496** [comment]
Someone modified the Desktop Notification part. Thats why this is not working anymore. I will fix both issues asap.

File: server/methods/saveUserPreferences.js (hunk 1 above)
Comment (aditya19496) [inline server/methods/saveUserPreferences.js]: Well I checked it personally. This change was not made by me. You can revert the changes by default for this part.

**Comment by aditya19496** [comment]
Its working perfectly when I commented that line of error. That line was neither added by me nor any changes were made by me.
Let me know what I can do.

**Comment by aditya19496** [comment]
Well i don't know what the issue is. I am opening new pull request with only the changes i will made. Hope it will work this time.
